x-pack/plugins/observability_solution/apm/public/components/app/service_inventory/index.tsx

```diff
@@ -342,6 +342,11 @@ export function ServiceInventory() {
                 !isTableSearchBarEnabled
               );
             }}
+            measureName="apm.services.table.services.list"
+            onMeasureComplete={
```

Comment:
Does this logic cover the full service page is rendered with all the necessary data the user is interested in? 

We need to measure the page readiness when this completion event can clearly convey when the page is meaningfully ready for the user.

Reply:
> Does this logic cover the full service page is rendered with all the necessary data the user is interested in?

it doesn't cover the _full services page_ but the _readiness_ of the component.  Once you navigate to the `/services` page and the component is mounted I set the start marker to measure the readiness of the component. 

In this case, when the data for the service table are fetched successfully it will mark the end of the measurement. 


**However, based on your comment** 
if there is ANY delay once you enter the page until the component is mounted it will not be captured.

I think we need an additional metric ex `actualDuration`  where we start measuring once you navigate to the page until the component is ready and fully rendered. 

 Is this close on how the page readiness is defined?

Reply:
I guess we need two things as we spoke on our last discussion. 

1. User centric page readiness event - Which will give a clear picture of when all the meaningful data is present for the user (Ex - In APM, this could be the when the data for traces/transactions is presented to the user). We should measure the actual render time of this page with all the relevant data pulled in which can be done with the same approach you have taken here using UserTiming API's. There is no way to accurately measure the actual render time, but can be done using a technique used in our RUM agent - https://github.com/elastic/apm-agent-rum-js/blob/58c8fa514f6cdc781c76cafb66421f8ecb0dfbab/packages/rum-core/src/common/after-frame.js#L40

2. Individual component level timings - This comes next after the fact we have identified the slowness on the page and we would like to drilldown which part of the page causes the slowness. I believe this PR is trying to solve this problem right now.

Reply:
Regarding 1:  I see your point. 

page readiness will be defined by the actual render time + meaningful data loaded successfully to the user.  
The meaningful data will be defined per page. it can be one or more data sources .

For instance, within the service list, the page defines "meaningful" as the collection of services (/services) **and** detailed statistics (/detailed_statistics.  In this example, the end marker will be marked by the time of the last succesful returned response. Did I understand it correctly ? 

Regarding 2: yes

Reply:
> page readiness will be defined by the actual render time + meaningful data loaded successfully to the user.
The meaningful data will be defined per page. it can be one or more data sources .

Agreed. 

> For instance, within the service list, the page defines "meaningful" as the collection of services (/services) and detailed statistics (/detailed_statistics. In this example, the end marker will be marked by the time of the last succesfull returned response

This should come from the product/engineers and similar to the metrics that have been around for years in the Web perf community as Time to First Meaningful Paint ([TTFMP](https://medium.com/airbnb-engineering/measuring-web-performance-at-airbnb-122da8d3ea3f#:~:text=for%20client%20transitions-,Time%20To%20First%20Meaningful%20Paint,-Time%20To%20First)). As you mentioned, if the services page readiness involves data from services collection and detailed statistics, that would be the indicator.

Reply:
We'll focus on the time to first meaningful paint  in this PR so I updated the whole PR